### ISSUE TITLE
Fails deploy when no Docker is installed without helpful error message

### ISSUE THREAD

**Issue by DerGut** [aws/copilot-cli#3169]
When running `copilot init` without having Docker installed it seems to crash without properly notifying the user about the reason.

At the Dockerfile prompt it logs `Docker command is not found; Copilot won't build from a Dockerfile.` and then continues the prompt with `What's the location of the image to use? [? for help]`. No Dockerfiles are suggested but the prompt accepts text.
After entering the name of the Dockerfile to use (manually), it correctly sets the `image.location` in the manifest file and then  continues with the rest of the setup, asking to deploy a test envirionment. Upon approval, it then fails after some time at the `Linking account 085424649605 and region eu-central-1 to application pearup.` step with:
```
✘ Failed to link account <ACCOUNT_ID> and region <REGION> to application <APP>.
✘ deploy env test to application <APP>: adding test environment resources to application: operation 5 for stack set <APP>-infrastructure failed
```

Trying to deploy manually, it complains that no environment exists. Then running `copilot env init`, it fails with the same message as above.

Instead of failing with the above error, I would expect the CLI to recognise that Docker is not installed and then stop trying to create an environment.

I am running copilot version `v1.13.0` on `macOS 12.1 `.

**Comment by efekarakus** [comment]
Hi @DerGut 👋 

Thanks for the detailed issue report! 

> After entering the name of the Dockerfile to use (manually), it correctly sets the image.location in the manifest file

Ah I see, I think the UX is confusing here the `location` field isn't meant to be a local path but the image name part of repository like `registry/repository[:tag]`. I've created #3173 which hopefully should improve both the UX and catch scenarios where a Dockerfile is passed as input.

> ```
>    ✘ deploy env test to application <APP>: adding test environment resources to application: operation 5 for stack set <APP>-infrastructure failed
> ```
Regarding the error above, I've tried to reproduce it but I don't think it's related to the Docker daemon not running. If you still have the failed application stack lingering around, would you mind going to the CloudFormation console and looking the reason why `operation 5` failed? https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/stacksets-update.html

**Comment by DerGut** [comment]
@efekarakus Thanks for the quick reply! 

I see, now it makes totally sense that `the location of the image` doesn't refer to the Dockerfile. Still feels good to have it clarified at little bit more!

I have replicated the scenario by renaming the `docker` binary to something that cannot be found by `copilot`. After running `copilot init` again with the actual image location it worked 🎉.

Doing the same thing but entering `Dockerfile` instead of a valid image url, is now succeeding too. This was definitely not true yesterday...

I see whether I can find anything that changed in my setup since yesterday. I believe I tried the whole thing three times and so it really appeared to be a consistent issue.
